### PR TITLE
Bugfix/21545 fix modbus disconnect

### PR DESCRIPTION
**Pull request by DC2-DanielKrueger**
https://hivemq.kanbanize.com/ctrl_board/57/cards/21545/
**Motivation**

Resolves #<issue>

**Changes**

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java [hunk 1]
```diff
@@ -199,138 +201,161 @@ public MonitoredPollingJob(final ProtocolAdapterPollingSampler sampler) {
...
             return notBefore > 0 || applicationErrorCount.get() > 0 || watchdogErrorCount.get() > 0;
         }
 
         @Override
         public void run() {
             long startedTimeMillis = System.currentTimeMillis();
             try {
-                if(notBefore > 0){
-                    if(System.currentTimeMillis() < notBefore){
+                if (notBefore > 0) {
```
Comment (sfrehse) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java:218]: Generally, this function reads very complex and is hard to follow. Do you think we can simplify that a bit?

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java [hunk 2]
```diff
@@ -199,138 +201,161 @@ public MonitoredPollingJob(final ProtocolAdapterPollingSampler sampler) {
...
-                    if(!continuing){
-                        if(log.isInfoEnabled()){
-                            log.info("Detected Bad System Process {} In Sampler {} - Terminating Process to Maintain Health ({}ms Runtime)",
-                                    errorCountTotal, sampler.getAdapterId(), System.currentTimeMillis() - startedTimeMillis);
+                    continuing = errorCountTotal <
+                            InternalConfigurations.ADAPTER_RUNTIME_WATCHDOG_TIMEOUT_ERRORS_BEFORE_INTERRUPT.get();
+                    if (!continuing) {
+                        if (log.isInfoEnabled()) {
+                            log.warn(
+                                    "Detected Bad System Process {} In Sampler {} - Terminating Process to Maintain Health ({}ms Runtime)",
```
Comment (sfrehse) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java:286]:
```suggestion
                                    "Detected bad system process {} in sampler {} - terminating process to maintain health ({}ms runtime)",
```

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java [hunk 3]
```diff
@@ -199,138 +201,161 @@ public MonitoredPollingJob(final ProtocolAdapterPollingSampler sampler) {
...
+                                    sampler.getAdapterId(),
+                                    System.currentTimeMillis() - startedTimeMillis);
                         }
                     } else {
-                        if(log.isDebugEnabled()){
-                            log.debug("Detected Bad System Process {} In Sampler {} - Interrupted Process to Maintain Health ({}ms Runtime)",
-                                    errorCountTotal, sampler.getAdapterId(), System.currentTimeMillis() - startedTimeMillis);
+                        if (log.isDebugEnabled()) {
+                            log.debug(
+                                    "Detected Bad System Process {} In Sampler {} - Interrupted Process to Maintain Health ({}ms Runtime)",
```
Comment (sfrehse) [inline hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java:294]: We may refactor this as well, since error message above is the same

File: hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java (hunk 1 above)
Comment (DC2-DanielKrueger) [reply hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/impl/ProtocolAdapterPollingServiceImpl.java:218]: absolutely, but not in this PR. This needs some real work and thoughts

**Review by sfrehse — APPROVED** [review]
